spec/lib/modules/report/collection_spec.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+describe 'TableExpression::Collection' do
+  # the correct order is A -> B -> C -> E -> D
```

Comment:
From a documentation perspective, maybe show the DAG:

A -> B -> C -> E -> D
\\---------------------^